base_layer/core/src/block.rs

```diff
@@ -52,6 +55,12 @@ impl Block {
     pub fn check_pow(&self) -> Result<(), TransactionError> {
         Ok(())
     }
+
+    /// This function will calculate the pow for the block and fill out the pow header field
+    pub fn calculate_pow(&mut self) -> Result<(), TransactionError> {
```

Comment:
I don't think you mean `calculate` pow. PoW is verified/validated, or else we're doing mining.

Reply:
No this is mining, its meant to call to the mining stuff when this is created.

Reply:
Block.rs shouldn't know about mining. We can leave this in for now, but this should be removed.